### PR TITLE
Enhancement/2038 user input admin screen

### PR DESCRIPTION
**Pull request by ryanwelcher**
## Summary

Addresses issue #2038

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: .storybook/webpack.config.js [hunk 1]
```diff
@@ -25,6 +31,13 @@ module.exports = async ( { config } ) => {
 	config.plugins = [
 		...config.plugins,
 		new MiniCssExtractPlugin(),
+		new FeatureFlagsPlugin(
+			flagsConfig,
+			{
+				modes: [ 'development', 'production' ],
+				mode: config.mode,
+			},
+		),
```
Comment (felixarntz) [inline .storybook/webpack.config.js:40]: Just to clarify, why is this needed in the Storybook Webpack?

File: assets/js/components/user-input/UserInputApp.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export default function UserInputApp() {
+	return featureFlags.widgets.userInput.enabled ? ( <div>TODO: Implement logic and UI.</div> ) : ( <div>You need a higher level of permission.</div> );
```
Comment (felixarntz) [inline assets/js/components/user-input/UserInputApp.js:20]: Just to make this more readable and prepare for future expansion, it would be better to break this into multiple lines. I think we should have an if clause `if ( ! featureFlags.widgets.userInput.enabled )`, and in that return `__( 'Something went wrong.', 'google-site-kit' )`. This part will be removed once the feature goes to production. Then the TODO div can be the default return value.

File: includes/Core/Admin/Screens.php [hunk 3]
```diff
@@ -383,6 +383,25 @@ private function get_screens() {
 			)
 		);
 
+		$screens[] = new Screen(
+			self::PREFIX . 'user-input',
+			array(
+				'title'            => __( 'User Input', 'google-site-kit' ),
+				'capability'       => Permissions::MANAGE_OPTIONS,
```
Comment (felixarntz) [inline includes/Core/Admin/Screens.php:390]: This also needs to pass the following so that the screen is not in the admin menu (per the ACs):
```suggestion
				'capability'       => Permissions::MANAGE_OPTIONS,
				'parent_slug'      => null,
```

File: stories/user-input.stories.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import { storiesOf } from '@storybook/react';
+
+/**
+ * Internal dependencies
+ */
+import UserInputApp from '../assets/js/components/user-input/UserInputApp';
+
+storiesOf( 'User Input', module )
+	.add( 'User Input Page', () => {
+		return <UserInputApp />;
```
Comment (felixarntz) [inline stories/user-input.stories.js:13]: While this component doesn't use the datastore yet since it's just scaffolded, let's already wrap this in a `WithDataRegistry` component, to which we can pass a default registry created via `createTestRegistry()`.

**Review by felixarntz — COMMENTED** [review]
This is pretty much good to be merged, just a few nit-pick suggestions and one question.

**Comment by felixarntz** [comment]
@ryanwelcher Can you also add a QA Brief for this one?

File: .storybook/webpack.config.js (hunk 1 above)
Comment (ryanwelcher) [reply .storybook/webpack.config.js:40]: @felixarntz this was added to allow access to the featureFlags. I have removed it and instead am setting the flag in the story.

File: includes/Core/Admin/Screens.php (hunk 3 above)
Comment (ryanwelcher) [reply includes/Core/Admin/Screens.php:390]: @felixarntz my mistake here, the IB says:
> args.parent_slug can be omitted as this will be null by default and will not appear in the admin menu.

I took omitted to mean "not added to the array" -  I have addressed this.

Comment (felixarntz) [reply includes/Core/Admin/Screens.php:390]: @ryanwelcher Oh good point - I must have missed that when reviewing the IB.

File: stories/user-input.stories.js (hunk 4 above)
Comment (ryanwelcher) [reply stories/user-input.stories.js]: @felixarntz I couldn't find any references to `WithDataRegistry`, so I am assuming that you meant `withTestRegistry` 😄

Comment (felixarntz) [reply stories/user-input.stories.js]: Oops, exactly 😆

**Review by felixarntz — APPROVED** [review]
Great!
